### PR TITLE
VS-58: Remapping Boolean Variables

### PR DESCRIPTION
**Pull request by Ravi-Raghavan**
Remapping Boolean Variables

### CODE REVIEW

File: tests/MongoDB.Analyzer.Tests.Common/DataModel/WeatherContainer.cs [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+namespace MongoDB.Analyzer.Tests.Common.DataModel
+{
+    public class WeatherContainer
+    {
+        public Weather weather { get; set; }
+    }
+    public class Weather
```
Comment (BorisDog) [inline tests/MongoDB.Analyzer.Tests.Common/DataModel/WeatherContainer.cs:21]: Empty line between classes.

File: tests/MongoDB.Analyzer.Tests.Common/MongoDB.Analyzer.Tests.Common.projitems [hunk 2]
```diff
@@ -33,5 +33,6 @@
     <Compile Include="$(MSBuildThisFileDirectory)TestCasesBase.cs" />
     <Compile Include="$(MSBuildThisFileDirectory)DataModel\User.cs" />
     <Compile Include="$(MSBuildThisFileDirectory)Location.cs" />
+    <Compile Include="$(MSBuildThisFileDirectory)DataModel\WeatherContainer.cs" />
```
Comment (BorisDog) [inline tests/MongoDB.Analyzer.Tests.Common/MongoDB.Analyzer.Tests.Common.projitems:36]: alphabetical order.

File: tests/MongoDB.Analyzer.Tests.Common/DataModel/WeatherContainer.cs [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+namespace MongoDB.Analyzer.Tests.Common.DataModel
+{
+    public class WeatherContainer
+    {
+        public Weather weather { get; set; }
```
Comment (BorisDog) [inline tests/MongoDB.Analyzer.Tests.Common/DataModel/WeatherContainer.cs:19]: All properties names capitalized.

File: tests/MongoDB.Analyzer.Tests.Common.TestCases/Builders/BuildersBasic.cs [hunk 4]
```diff
@@ -320,5 +320,36 @@ public void Multiple_expression_variables_declaration_and_reassignment()
...
         }
+
+        [BuildersMQL("{ \"isRaining\" : { \"$exists\" : isExist } }")]
+        [BuildersMQL("{ \"isRaining\" : isExist }")]
+        [BuildersMQL("{ \"isRaining\" : { \"$in\" : [isExist, isExist, isExist] } }")]
+        [BuildersMQL("{ \"boolArray\" : true }")]
+        [BuildersMQL("{ \"isRaining\" : { \"$ne\" : isExist } }")]
+        public void Filter_with_boolean_variables()
+        {
+            var isExist = true;
```
Comment (BorisDog) [inline tests/MongoDB.Analyzer.Tests.Common.TestCases/Builders/BuildersBasic.cs:331]: isExist => exists.

File: tests/MongoDB.Analyzer.Tests.Common/DataModel/WeatherContainer.cs (hunk 1 above)
Comment (Ravi-Raghavan) [reply tests/MongoDB.Analyzer.Tests.Common/DataModel/WeatherContainer.cs:21]: Done

File: tests/MongoDB.Analyzer.Tests.Common/DataModel/WeatherContainer.cs (hunk 3 above)
Comment (Ravi-Raghavan) [reply tests/MongoDB.Analyzer.Tests.Common/DataModel/WeatherContainer.cs:19]: Done

File: tests/MongoDB.Analyzer.Tests.Common.TestCases/Builders/BuildersBasic.cs (hunk 4 above)
Comment (Ravi-Raghavan) [reply tests/MongoDB.Analyzer.Tests.Common.TestCases/Builders/BuildersBasic.cs:331]: Done

File: tests/MongoDB.Analyzer.Tests.Common/MongoDB.Analyzer.Tests.Common.projitems (hunk 2 above)
Comment (Ravi-Raghavan) [reply tests/MongoDB.Analyzer.Tests.Common/MongoDB.Analyzer.Tests.Common.projitems]: Done

**Comment by BorisDog** [comment]
This feature requires more research, and maybe different implementation.

**Pull request closed without merging**